lib/Net/SAML2/IdP.pm

```diff
@@ -223,22 +227,42 @@ Called after the object is created to validate the IdP using the cacert
 
 =cut
 
-sub BUILD {
-    my($self) = @_;
-
-    if ($self->cacert) {
-        my $ca = Crypt::OpenSSL::Verify->new($self->cacert, { strict_certs => 0, });
-
-        for my $use (keys %{$self->certs}) {
-            my $cert = Crypt::OpenSSL::X509->new_from_string($self->certs->{$use});
-            ## BUGBUG this is failing for valid things ...
-            eval { $ca->verify($cert) };
-            if ($@) {
-                warn "Can't verify IdP '$use' cert: $@\n";
+around BUILDARGS => sub {
+    my $orig = shift;
+    my $self = shift;
+
+    my %params = @_;
+
+    if ($params{cacert}) {
+        my $ca = Crypt::OpenSSL::Verify->new($params{cacert}, { strict_certs => 0, });
+
+        my $verified = 0;
+        my $error = "";
```

Comment:
I feel like $error should be an array in this context. Each cert can have a different error. At line 252 we just replace the error with the latest and greatest error we get.